### ISSUE TITLE
[Bug]: Download Stuck

### ISSUE THREAD

**Issue by jay-oswald** [tubearchivist/tubearchivist#715]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Unraid

## Describe the bug
I have a download that appears "stuck". In the GUI its stuck and not moving, and I can't remove it it, or download other files
## Steps To Reproduce
I had files downloading in the background, but I did an unraid update. My best guess is unraid requested a shutdown, tubearchivest was waiting for the download to finish, unraid did a forceful shutdown after its graceful timer was reached, leaving it in this stuck state.

I think that the temporary files are stored outside of persistant volumes, so even if it did gracefully shutdown, it thinks the partial file is there, and its not, so it can't resume the download, which is why its getting a `FileNotFoundError` error.

## Expected behavior
`FileNotFoundError` errors should be detected, and since it thinks the file is there and its not, the download should just restart, rather than trying to resume, which is what I assume its doing

### Relevant log output

```shell
[2024-05-11 19:47:00,009: INFO/MainProcess] Task download_pending[9f270069-72d8-43cf-9448-e8404684bd96] received
[2024-05-11 19:47:00,009: WARNING/ForkPoolWorker-14] download_pending create callback
[2024-05-11 19:47:00,172: WARNING/ForkPoolWorker-14] iVzQsFWBXIQ: Downloading video
[2024-05-11 19:47:01,752: WARNING/ForkPoolWorker-14] WARNING: [youtube] Skipping player responses from android clients (got player responses for video "aQvGIIdgFDM" instead of "iVzQsFWBXIQ")
[2024-05-11 19:47:38,358: WARNING/ForkPoolWorker-14] iVzQsFWBXIQ: get metadata from youtube
[2024-05-11 19:47:39,993: WARNING/ForkPoolWorker-14] WARNING: [youtube] Skipping player responses from android clients (got player responses for video "aQvGIIdgFDM" instead of "iVzQsFWBXIQ")
[2024-05-11 19:47:42,001: WARNING/ForkPoolWorker-14] UCLe_q9axMaeTbjN0hy1Z9xA: get metadata from es
[2024-05-11 19:47:42,027: WARNING/ForkPoolWorker-14] 9f270069-72d8-43cf-9448-e8404684bd96 Failed callback
[2024-05-11 19:47:42,028: ERROR/ForkPoolWorker-14] Task download_pending[9f270069-72d8-43cf-9448-e8404684bd96] raised unexpected: FileNotFoundError()
Traceback (most recent call last):
  File "/root/.local/lib/python3.11/site-packages/celery/app/trace.py", line 477, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/celery/app/trace.py", line 760, in __protected_call__
    return self.run(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/home/tasks.py", line 208, in download_pending
    videos_downloaded = downloader.run_queue(auto_only=auto_only)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/home/src/download/yt_dlp_handler.py", line 182, in run_queue
    vid_dict = index_new_video(
               ^^^^^^^^^^^^^^^^
  File "/app/home/src/index/video.py", line 409, in index_new_video
    video.build_json()
  File "/app/home/src/index/video.py", line 154, in build_json
    self.add_player(media_path)
  File "/app/home/src/index/video.py", line 255, in add_player
    vid_path = media_path or self.build_dl_cache_path()
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/home/src/index/video.py", line 251, in build_dl_cache_path
    raise FileNotFoundError
FileNotFoundError
```

### Anything else?

![image](https://github.com/tubearchivist/tubearchivist/assets/16188552/2ec915f8-6a3a-4fab-81c9-147c6a6633cb)

It is stuck with a percent, download size and speed, so I assume it is stuck in the database. Pressing ignore on the video on the download list lets it continue to download other videos. I have not tried to re-download the video yet, I would like it to continue downloading the queue overnight, will retry that specific video when the download queue has caught up (only recently installed, sitll doing an initial sync)

**Comment by bbilly1** [comment]
There are a lot of scenarios that will raise the `FileNotFoundError` error, all outside of the control of this app. If your OS auto updates, you'll want to configure that so this doesn't do this.

That can also happen with things like drive issues, e.g. no storage left to merge the different streams. In that case, you wouldn't want to just download it again.

Restart should bring you back to a clean state and you should be able to download it again.

**Comment by jay-oswald** [comment]
Thanks for the reply, my docker disk on unraid did hit 95% disk, I freed up room, and restartde, but unsure on the order I did it in. I must have restarted, then cleared up room, and doing it in the other order would have worked.

The update wasn't automatic, I just let it gracefully shutdown containers, which is usually fine, and this may be un-releated anyway.

If it was just a disk space issue it should have cleared itself though? As I took disk usage from 95% down to 70%, well under the 90% threshold?
